samples/tfm/tfm_secure_peripheral/README.rst

```diff
@@ -0,0 +1,166 @@
+This is done with a bitmask, for example to assign GPIO pin 23 as secure:
+
+.. code-block:: none
+
+	CONFIG_NRF_GPIO0_PIN_MASK_SECURE=0x00800000
+
+In addition, the peripheral must be assigned to the specific partition in the partition manifest file:
+
+.. code-block:: none
+
+        "mmio_regions": [
+                {
+                        "name": "TFM_PERIPHERAL_TIMER1",
+                        "permission": "READ-WRITE"
+                },
+        ]
+
+Secure interrupt
+================
+
+If the secure peripheral generates interrupts, the interrupt needs to be integrated with TF-Ms interrupt mechanism.
+The interrupt source and handling type needs to be defined in the partition manifest file:
+
+.. code-block:: none
+
+        "irqs": [
+                {
+                        "source": "TFM_TIMER1_IRQ",
+                        "name": "TFM_TIMER1_IRQ",
+                        "handling": "FLIH"
+                },
+        ]
+
+If the type of interrupt handling is defined as the First Level Interrupt Handling (FLIH), then the FLIH handler needs to be defined.
+The return value specifies if the Second Level Interrupt Handling signal should be asserted or not.
+
+.. code-block:: c
+
+        psa_flih_result_t tfm_timer1_irq_flih(void)
+        {
+                /* Application specific handling */
+
+                if (condition) {
+                        return PSA_FLIH_SIGNAL;
+                } else {
+                        return PSA_FLIH_NO_SIGNAL;
+                }
+        }
+
+If the type of interrupt handling is defined as the Second Level Interrupt Handling (SLIH), or the FLIH handler specifies the signal to be raised, the interrupt is sent to the partition as a signal.
+How to clear an interrupt signal depends on the type of interrupt handling.
+
+.. code-block:: c
+
+        if (signals & TFM_TIMER1_IRQ_SIGNAL) {
+                /* Application specific handling */
+
+                /* FLIH: Reset signal */
+                psa_reset_signal(TFM_TIMER1_IRQ_SIGNAL);
+        }
+
+        if (signals & TFM_SPIM3_IRQ_SIGNAL) {
+                /* Application specific handling */
+
+                /* SLIH: End of Interrupt signal */
+                psa_eoi(TFM_SPIM3_IRQ_SIGNAL);
+        }
+
+.. note::
+
+        TF-M interrupt signals only assert the signal but do not schedule the partition to run.
+        In cases where the interrupt signal is preempting the non-secure execution, the interrupt signal is not processed until the next time the partition is scheduled to run.
+        The sample demonstrates a workaround for this limitation by triggering an EGU0 interrupt in the non-secure firmware which calls the secure partition to process the interrupt signals.
+
+
+Building and Running
+********************
+
+.. |sample path| replace:: :file:`samples/tfm/tfm_secure_peripheral`
+
+.. include:: /includes/build_and_run.txt
+
+Testing
```

Comment:
Why is the logic analyzer needed? Can't you verify the behavior by looking at this console output?

Reply:
It's not required, the sample demonstrates using SPI in secure. A logic analyzer can be used to see some pin toggling of "random" data.

Will have to check with the testers how best to proceed with this.